### ISSUE TITLE
No option to search videos

### ISSUE THREAD

**Issue by vaibhavsingh97** [fossasia/query-server#320]
With reference to #58 

**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
There is no option to search videos in Search engines

**Expected behavior:**
There should be option to search videos

**Open To claim**

**Comment by raju249** [comment]
@vaibhavsingh97 ,

I can take this up if you are busy with something else :smile: 
let me know what you think ?

**Comment by vaibhavsingh97** [comment]
@raju249 Are you working on the issue?

**Comment by umangahuja1** [comment]
Assign this issue to me since nobody is working as of now and I am willing to take this up. @vaibhavsingh97

**Comment by vaibhavsingh97** [comment]
@umangahuja1 I can't assign the issue to you but you can work on this issue.
Thanks

**Comment by umangahuja1** [comment]
Okay since you said to get assigned but I don't know how to do it. I am starting it then. 


![image](https://user-images.githubusercontent.com/22851705/34482490-560782a4-efde-11e7-8b69-daa6233ad1d7.png)

Just for confirmation, this feature intend to get these title and url from videos section?

Like https://www.icc-cricket.com/womens-world-cup/video/440702 for the first item.

**Comment by vaibhavsingh97** [comment]
Yes @umangahuja1 So what we discussed in gitter to have one extra attribute type which contains `url`, `image` and `video` and based on the type and format. result will be displayed.

**Comment by umangahuja1** [comment]
Great so for now I will start will google, but I think this will be done for all those search engines that offer images and videos.

So most probably all the changes will be made in their respective files (e.g for google google.py) and other functions will be added to scrape accordingly.

Is this what you are doing in image feature?

**Comment by bhaveshAn** [comment]
@umangahuja1 Since this issue has been taken too long, without any progress, so please work on this a little quick, with regular update as this need regular discussions on the approach.

Also, I have fixed half of it. So if you aren't fully comfortable with the approach then let me make a pull on Video search in 3-4 days or so. and then you can switch to other file types like Image search etc (As the approach will be same for both the searches).
I think it would be better.

**Comment by umangahuja1** [comment]
What have you done, if anything?
Because it's not a bug, it's a feature so there is nothing to be fixed I suppose.

Better to discuss in chat or wherever with others too if you have done something because I took it today itself. Switch to gitter.

**Comment by bhaveshAn** [comment]
All I mean I am about to give the PR on this in 3-4 days. So please don't work on this.
@vaibhavsingh97 You can assign it to me as this is definitely not a beginners issue, advanced one as all the enhancement on all the supported search engines.
And @umangahuja1 Please start the project from some beginners issue. Or check if https://github.com/fossasia/query-server/issues/321 is available as it is also not assigned.
Maybe you can take #368  #265

**Comment by vaibhavsingh97** [comment]
@bhaveshAn IMO, i think @umangahuja1 should get a fair chance. Let him try. I know it's not an beginner issue but what's wrong in giving a chance. we can help him if he got stuck somewhere. It's good that you want to work on this issue but @umangahuja1 asked to claim first so h should get the fair chance. Thanks

**Comment by harshit98** [comment]
@bhaveshAn @vaibhavsingh97 What's the matter?

@umangahuja1 Hi. You may please proceed to solve this issue and take your time. If you need any help, chat is the best way to communicate!

**Comment by umangahuja1** [comment]
Here is the screenshot of video feature

![image](https://user-images.githubusercontent.com/22851705/34524892-246f2eaa-f0c3-11e7-8d63-886e9966048b.png)

![image](https://user-images.githubusercontent.com/22851705/34524884-193c6a02-f0c3-11e7-8332-0489c6be871b.png)

I will add front end and make it working soon.

**Comment by Remorax** [comment]
@vaibhavsingh97 @umangahuja1 @harshit98 Can I work on the other search engines? It would be better if the work is divided, right?

**Comment by umangahuja1** [comment]
I think that will be a great idea. I am working on bing so you can choose any other. We can pace up things this way. But for that it would be better to merge the PR I sent since it will be used in all other engined.

What do you all say?

**Comment by Remorax** [comment]
That's cool @umangahuja1 😄 
I'll be making a PR within the next 2 days 😄

**Comment by umangahuja1** [comment]
Wait let the google search PR get merged since it has front end part. Also generalised.py and one more file has some changes that will be used in all.

**Comment by bhaveshAn** [comment]
Okay, so I will also be working for video search in Baidu and Ask.

**Comment by Remorax** [comment]
Cool, I'll be working on Yahoo and DuckDuckGo.

**Comment by Remorax** [comment]
@umangahuja1 Sure we'll wait for your PR to get merged, don't worry 😄 .
Okay @bhaveshAn ?

**Comment by Remorax** [comment]
@bhaveshAn Let's divide the work. I'll be working on Mojeek, Ask and DuckDuckGo. Can you work on the rest? Thanks 😄

**Comment by bhaveshAn** [comment]
Mojeek is not supporting Image/Video search check it out https://www.mojeek.co.uk/search?q=fossasia
I am working on Ask Video search as claimed previously https://github.com/fossasia/query-server/issues/320#issuecomment-357495815

You can proceed with Duckduckgo

**Comment by Remorax** [comment]
@bhaveshAn you had claimed Baidu and Ask but you made PRs for Parsijoo and Bing 😆 
I was actually in the middle of working on Bing, so idk what to say to that. But yeah, I'll work on DuckDuckGo then.

**Comment by bhaveshAn** [comment]
Yeah, earlier, I had claimed Ask and Baidu, and you claimed Yahoo (For which made the PR) and Duckudckgo, but there was no discussion on Parsijoo and Bing, thats why I made the PR, suddenly, since no one here claimed rest out of `Ask, Baidu    Yahoo, DuckDuckGo`

**Comment by Remorax** [comment]
That's true but you should probably have worked on Baidu and Ask first and then work on Bing and Parsijoo after asking others and making sure no one else is working on it. Peace 😄

**Comment by bhaveshAn** [comment]
@Remorax reminder

**Comment by Remorax** [comment]
Oh, yeah sure @bhaveshAn work on DuckDuckGo and I'll work on Baidu 😄 
And if anyone wants to work on any search engine, please post here and confirm whether somebody else is working on it or not, so as to not waste anybody's time. Thanks 😄
